shakenfist/external_api/artifact.py

```diff
@@ -48,6 +48,41 @@ class ArtifactEndpoint(api_base.Resource):
     def get(self, artifact_uuid=None, artifact_from_db=None):
         return artifact_from_db.external_view()
 
+    @jwt_required
+    def delete(self, artifact_uuid=None):
+        '''Delete an artifact from the cluster
+
+        Artifacts can only be deleted from the system if they are not in use.
+        The actual deletion of the on-disk files is left to the cleaner daemon.
+
+        It is acknowledged that there is a potential race condition between the
+        check that an artifact is not in use and the marking of the artifact as
+        deleted. This is only caused by a user simultaneously deleting an
+        artifact and attempting to start a VM using it. It is recommended that
+        the user does not do that.
+        '''
```

Comment:
The quote thing again.

Reply:
Also, nice passive aggresive comment. That's a keeper.

Reply:
The alternative was a major code change in instance create to get and hold locks during instance creation.